### ISSUE TITLE
Question: thoughts behind this api

### ISSUE THREAD

**Issue by infinite-sahil** [landawn/AbacusUtil#22]
Hi @landawn 

Just a random question came in my mind after working on your api from last 3 monhts,
This api is no doubt beautifully written, so few question came to my mind, i though i should ask you.

1. What is the though process/idea behind creating this project.
2. How long it took you to write this api.

you worked delicately on this or just to bring the change you developed this beautiful api.

Thank you. have a good day

**Comment by landawn** [comment]
That's a long story... short answer to your question: Why&How?
I know there are some very popular & widely used frameworks/libraries. However pretty few I like. (simple input for the reason: not good enough by my standards). 
I try my best to make this framework as good as possible. There are more than 10K commits in my local repository. Most of time, one commit means one or more changes/improvements/ideas. Mostly i work on it in my spare time and it's a really long journey, which means a lot of time/efforts/thoughts.
I created this project from scratch. but there're also a lot of ideas/codes I learned/copied(under Apache License v2) from other people/frameworks.

I'm glad to hear you're enjoy using it!
